notification-hubs-sdk/src/main/java/com/microsoft/windowsazure/messaging/notificationhubs/DebounceInstallationAdapter.java

```diff
@@ -0,0 +1,61 @@
+package com.microsoft.windowsazure.messaging.notificationhubs;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.microsoft.windowsazure.messaging.R;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public class DebounceInstallationAdapter implements InstallationAdapter {
+
+    private static final String PREFERENCE_KEY = "recentInstallation";
+    private final ScheduledExecutorService mScheduler = Executors.newScheduledThreadPool(1);
+    private InstallationAdapter mInstallationAdapter;
+    private long mInterval;
+    private ScheduledFuture<?> mSchedFuture;
+    private SharedPreferences mPreferences;
+
+    public DebounceInstallationAdapter(InstallationAdapter installationAdapter) {
+        this(installationAdapter, 2000);
+    }
+
+    public DebounceInstallationAdapter(InstallationAdapter installationAdapter, long interval) {
+        super();
+        this.mInstallationAdapter = installationAdapter;
+        mInterval = interval;
+    }
+
+    private void setPreferences(Context context) {
```

Comment:
The name `setPreferences` is a bit confusing since this isn't a normal "set" method. Maybe `initPreferences` would be better?

Reply:
In some of the later work, I started putting this in the constructor and not allowing it to get set later. I may just do that here too.

Reply:
Fixed in cc26243bef08124b824e98f181770a55656b92cc